### PR TITLE
odb: def parsing and writing of MASK property on VIAs

### PR DESCRIPTION
**Pull request by gadfort**

### CODE REVIEW

**Comment by maliberty** [comment]
def_parser (tcl) *ERROR*

File: src/odb/test/data/design58.def
Comment (maliberty) [inline src/odb/test/data/design58.def:1]: is there a test for defout?

File: src/odb/src/db/dbWireCodec.cpp [hunk 1]
```diff
@@ -642,6 +642,32 @@ void dbWireEncoder::clearColor()
...
+  for (const auto color : {bottom_color, cut_color, top_color}) {
+    if (color > 3) {
+      utl::Logger* logger = _wire->getImpl()->getLogger();
+      logger->error(
+          utl::ODB, 1103, "Mask color: {}, but must be between 0 and 3", color);
+    }
+  }
+
+  // encode as XX BB CC TT
+  const uint8_t mask_color = bottom_color << 4 | cut_color << 2 | top_color;
```
Comment (QuantamHD) [inline src/odb/src/db/dbWireCodec.cpp:660]: You should just make a struct with bit fields, and then cast it to uint8 in the WOP. It'll save you from having to do the bit twiddling manually

Comment (gadfort) [reply src/odb/src/db/dbWireCodec.cpp:660]: I would be all for that, but I haven't tried that before and a quick stab at it didn't work, so I'll probably need some help as to how to do that.

File: src/odb/test/data/design58.def
Comment (gadfort) [reply src/odb/test/data/design58.def:1]: added a test for `write_def58`

File: src/odb/src/db/dbWireCodec.cpp (hunk 1 above)
Comment (maliberty) [reply src/odb/src/db/dbWireCodec.cpp:660]: @QuantamHD I'm inclined to merge and you can PR your idea as an improvement.

**Review by maliberty — APPROVED** [review]

**Pull request merged**